src/irmin-pack/unix/stats.ml

```diff
@@ -171,13 +171,72 @@ module Index = struct
   let export m = Metrics.state m
 end
 
-type t = { pack_store : Pack_store.stat; index : Index.stat }
+module File_manager_stats = struct
+  type Metrics.origin += File_manager_stats
 
-let s = { pack_store = Pack_store.init (); index = Index.init () }
+  type t = {
+    mutable dict_flushes : int;
+    mutable suffix_flushes : int;
+    mutable index_flushes : int;
+    mutable auto_dict : int;
+    mutable auto_suffix : int;
+    mutable auto_index : int;
+    mutable flush : int;
+  }
+  [@@deriving irmin]
+
+  (* NOTE return a new instance each time, since fields are mutable *)
+  let init' () =
+    {
+      dict_flushes = 0;
+      suffix_flushes = 0;
+      index_flushes = 0;
+      auto_dict = 0;
+      auto_suffix = 0;
+      auto_index = 0;
+      flush = 0;
+    }
+
+  (* type [stat] is an abstract type in stats.mli; FIXME what is it for? *)
```

Comment:
https://github.com/mirage/irmin/blob/main/src/irmin/metrics.mli#L25 - and there are more details here https://github.com/mirage/irmin/pull/1817#issuecomment-1111121636

Reply:
Yes, I sort-of understand what metrics adds. But this is a question about "stat": why have type stat at all? It doesn't really seem to add anything.